pulser/tests/test_paramseq.py

```diff
@@ -29,7 +29,7 @@
 def test_var_declarations():
     sb = Sequence(reg, device)
     assert sb.declared_variables == {}
-    var = sb.declare_variable("var")
+    var = sb.declare_variable("var", size=1)
```

Comment:
Maybe you could add a test where the variable is declared with `size=None` too

Reply:
This corresponds to the line `var = sb.declare_variable("var")` since `size` is defaulted to `None`. Do you think I should add a test where it is explicitely `None` too?

Reply:
Yes, but you changed it to have `size=1` so now I don't know if there is one where size was not specified (ie defaulted to `None`). No need for a test where `None` is explicitly stated

Reply:
I meant that the line `var = sb.declare_variable("var")` is already present in other parts of the test file. For example, on line 134 in `test_build()`

Reply:
Sure, of course. But I think it would be worth it to have the checks that are done after this declaration for this case too. For example:
```python
var_ = seq.declare_variable("var_")
assert sb.declared_variables["var_"] == var_
assert isinstance(var_, _VariableItem)
```
You see what I mean?

Reply:
Yes, thanks for your remark, I have applied it, although a downside of this approach in my sense is that it seems to be testing the implementation instead of the behavior.